.github/workflows/publish.yml

```diff
@@ -1,28 +1,40 @@
-name: publish
+name: Publish
 
 on:
   release:
-    types: [created]
+    types: [released]
+
+concurrency:
+  group: ci-publish
+  cancel-in-progress: true
 
 jobs:
   publish:
     runs-on: ubuntu-latest
 
+    permissions:
```

Comment:
Why should we define `permissions` explicitly?

Reply:
This is necessary for the "provenance" option to work correctly
https://docs.npmjs.com/generating-provenance-statements